custom_components/sat/minimum_setpoint.py

```
@@ -11,46 +17,70 @@ class MinimumSetpoint:
 
-    def warming_up(self, return_temperature: float) -> None:
-        if self.base_return_temperature is not None and self.base_return_temperature > return_temperature:
+    def warming_up(self, boiler_state: BoilerState) -> None:
+        if self._base_return_temperature.value is not None and self._base_return_temperature.value == boiler_state.return_temperature:
             return
 
-        # Use the new value if it's higher or none is set
-        self.base_return_temperature = return_temperature
-        _LOGGER.debug(f"Higher temperature set to: {return_temperature}.")
+        self._base_return_temperature = update_state(previous=self._base_return_temperature, new_value=boiler_state.return_temperature)
+        _LOGGER.debug(f"New temperature set to: %.1f°C.", self._base_return_temperature.value)
```

Ruff (0.14.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)